packages/eslint-plugin/src/rules/sort-interface-members.ts

```diff
@@ -0,0 +1,212 @@
+/**
+ * @fileoverview Forbids unsorted interface members
+ */
+
+import * as util from '../util';
+import {
+  Identifier,
+  TSCallSignatureDeclaration,
+  TSConstructSignatureDeclaration,
+  TSIndexSignature,
+  TSMethodSignature,
+  TSPropertySignature,
+  TSTypeAnnotation,
+  TSTypeReference,
+  TypeElement,
+} from '@typescript-eslint/typescript-estree/dist/ts-estree/ts-estree';
+
+type Options = [];
+type MessageIds = 'notSorted';
+
+function isPropertySignature(
+  member: TypeElement,
+): member is TSPropertySignature {
+  return (<TSPropertySignature>member).type === 'TSPropertySignature';
```

Comment:
Two things:
1) you don't need the type cast, as all of the members of the type union include the `type` property.
2) You should avoid the raw strings, use `AST_NODE_TYPES` instead.

```ts
  return member.type === AST_NODE_TYPES.TSPropertySignature;
```

However this function is kind of useless because the check is so simple.
Additionally technically it'll be slower at runtime because the engine will have to create a new heap in order to call the function which only lasts for 1 operation.